### PR TITLE
Refs #29483 - remove unecessary use of implicit scopes

### PR DESCRIPTION
**Pull request by ezr-ondrej**
This is not exhaustive list, probably there are other places we will need to address, this is example and it's the low hanging fruit before we will be forced to do it for Rails 6.1 :upside_down_face:

### CODE REVIEW

**Comment by mmoll** [comment]
Only one still on:
```
- message: Class level methods will no longer inherit scoping from `create!` in Rails
    6.1. To continue using the scoped relation, pass it into the block directly. To
    instead access the full set of models, as Rails 6.1 will, use `Puppetclass.unscoped`,
    or `Puppetclass.default_scoped` if a model has default scopes.
  callstack: app/services/authorizer.rb:78:in `find_collection'

**Comment by ezr-ondrej** [comment]
I think there was anotherone as well. But if it should be just one, let me take a look! :eye:

File: app/services/authorizer.rb [hunk 1]
```diff
@@ -70,7 +70,7 @@ def find_collection(resource_class, options = {})
       end
     else
       # build regular filtered scope for "resource_class"
-      scope = resource_class
+      scope = resource_class.default_scoped
```
Comment (ezr-ondrej) [inline app/services/authorizer.rb]: gotcha!

**Comment by ezr-ondrej** [comment]
Almost confident to mark this `Fixes` :smirk: WDYT @mmoll?

**Comment by mmoll** [comment]
tests fail :broken_heart:

**Comment by ezr-ondrej** [comment]
Right, this is the place, where we abuse the class scope inheritance. I'll create followup PR, but it can wait for Rails 6.1, so we can get this in now and try to refactor the `Authorizer` a bit before we hit 6.1

File: app/services/authorizer.rb (hunk 1 above)
Comment (ezr-ondrej) [reply app/services/authorizer.rb]: Reverted, that won't be as easy.

**Review by tbrisker — APPROVED** [review]

**Comment by mmoll** [comment]
Katello test failures unrelated, AFAICT.

**Comment by mmoll** [comment]
merged, díky @ezr-ondrej!

**Pull request merged**